File: certificates/problem-solving-basic/vowel-substring/findSubstring-2.js
function findSubstring(s, k) {
  let vowels = ["a", "e", "i", "o", "u"];
  let count = 0;
  let ans = 0;

  let key = {};
  for (let x of vowels) {
    key[x] = 1;
  }

  for (let i = 0; i < k; i++) {
    count += (key[s[i]] || 0);
  }
  let max = count;

  for (let i = k; i < s.length; i++) {
    count += (key[s[i]] || 0);
    count -= (key[s[i - k]] || 0);
    if (count > max) {
      max = count;
      ans = i - k + 1;
    }
  }
  if (max > 0) {
    return s.slice(ans, ans + k);
  }
  return "Not found!";


}

File: certificates/problem-solving-basic/vowel-substring/findSubstring-2.c
#include <stdio.h>
#include <string.h>

char* findSubstring(char* s, int k)
{
  char *result = (char *)malloc((k + 1) * sizeof(char));
  int key[256] = {};
  char vowels[] = "aeiou";
  for (int i = 0; i < strlen(vowels); i++)
  {
    key[vowels[i]] = 1;
  }

  int count = 0;
  int ans = 0;

  for (int i = 0; i < k; i++)
  {
    count += key[s[i]];
  }
  int max = count;
  for (int i = k; i < strlen(s); i++)
  {
    count += key[s[i]];
    count -= key[s[i - k]];

    if (count > max)
    {
      max = count;
      ans = i - k + 1;
    }
  }

  if (max > 0)
  {
    strncpy(result, s + ans, k);
    result[k] = '\0';
    return result;
  }

  free(result);
  return "Not found!";
}